src/openms/include/OpenMS/METADATA/SpectrumSettings.h

```diff
@@ -78,6 +79,14 @@ namespace OpenMS
     ///sets the spectrum type
     void setType(SpectrumType type);
 
+    /// @brief sets the IMFormat of the spectrum
+    /// @param im_type
+    void setIMFormat(const IMFormat& im_type);
+
+    /// @brief returns the IMFormat of the spectrum
+    /// @return IMFormat of the spectrum
+    IMFormat getIMFormat() const;
```

Comment:
not entirely happy here, since the user may not be aware that the value is not set by default and needs to be computed using IMTypes. Maybe add a comment

Reply:
true but maybe we can set this automatically in the common FileHandlers. will add a comment